### PR TITLE
Update tutorials

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Update and improve the two introduction tutorials that we also use for our courses. The goal of this is to be able to use identical notebooks in the course since maintaining separate variants is not sustainable in the long run.

See also https://github.com/ess-dmsc-dram/python-course-ikon/tree/master/notebooks/7_scipp.

### CODE REVIEW

File: docs/tutorials/working-with-masks.ipynb [hunk 1]
```diff
@@ -117,20 +129,26 @@
    "outputs": [],
    "source": [
     "tof = data.coords['tof']\n",
-    "data.masks['prompt-pulse'] = (tof['tof',:-1] > 17500.0 * sc.units.us) & (tof['tof',1:] < 19000.0 * sc.units.us)\n",
-    "sc.plot({'before':counts, 'after':sc.sum(data, 'tof')})"
+    "data.masks['prompt-pulse'] = (tof['tof',:-1] > prompt_start) & (tof['tof',1:] < prompt_stop)\n",
```
Comment (nvaytet) [inline docs/tutorials/working-with-masks.ipynb:132]: Why are we doing the masking a second time here, if it has already been done 2 cells above?

File: docs/tutorials/exploring-data.ipynb [hunk 2]
```diff
@@ -73,14 +78,28 @@
...
    "metadata": {},
    "source": [
-    "### Step 2: Plot the data\n",
+    "Whenever you get stuck with one of the exercises below we recommend consulting the HTML representations of the objects you are working with."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "jp-MarkdownHeadingCollapsed": true,
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply docs/tutorials/exploring-data.ipynb:87]: It is supposed to collapse everything until the next markdown heading, i.e., the whole section.

**Comment by SimonHeybrock** [comment]
Thanks, I believe I addressed your comments. The markdown heading collapse appears to be partially broken. I suggest we wait for until if upstream will fix this.

**Comment by SimonHeybrock** [comment]
I have opened https://github.com/jupyterlab/jupyterlab/issues/11119 upstream.

**Review by nvaytet — APPROVED** [review]

**Pull request merged**